[DOCS] Adds anomaly job types docs to ML book (#2307)
elastic/stack-docs: docs/en/stack/ml/anomaly-detection/ml-ad-job-types.asciidoc

[[ml-anomaly-detection-job-types]]
= {anomaly-detect-cap} job types
++++
<titleabbrev>Job types</titleabbrev>
++++

:keywords: {ml-init}, {stack}, {anomaly-detect}

:description: The description of the different anomaly detection job types. 

{anomaly-jobs-cap} have many possible configuration options which enable you to 
fine-tune the jobs and cover your use case as much as possible. This page 
provides a quick overview of the different types of {anomaly-jobs} and their 
capabilities. The currently available job types are:

* single metric jobs,
* multi-metric jobs,
* population jobs,
* categorization jobs,

> * rare jobs.

Reviewer: As these are the types of wizard in the ML UI, I think "advanced jobs" should also be in the list.

event is anomalous if the request rate of an IP address is unusually high or low 
compared to the request rate of all IP addresses in the population. The 
population job builds a model of the typical number of requests for the IP 
addresses collectively and compares the behavior of each IP address against that 
collective model to detect outliers. 

[discrete]
[[categorization-jobs]]
## Categorization jobs

Categorization jobs cluster similar text values together, classify them into 
categories and detect anomalies within the categories. Categorization works best 
on machine-written text like log messages that typically contains repeated 
strings of text; it does not work well on human-generated text because of its 
high variability.

The model learns the normal volume and pattern of a category over time so the 
job can detect anomalous behavior, such as an unusual number of events in a 
category by using the `count` function or messages that rarely occur by using 
the `rare` function.

[discrete]
[[rare-jobs]]
## Rare jobs

Rare {anomaly-jobs} detect rare occurrences in time series data. Rare jobs use 

> the `rare` or `frequ_rare` functions and detect such events in populations as 

Reviewer: ```suggestion
the `rare` or `freq_rare` functions and detect such events in populations as 
```

well. A _rare_ job finds events in simple time series data that occur rarely 
compared to what the model observed over time. A _rare in a population_ job 
finds members of a population that have rare values over time compared to the 
other members of the population. The _frequently rare in a population_ job 
detects such rare events that frequently occur for a member of a population 
compared to other members. As an example, a job models URI paths and client IP 
interactions and defines a rare URI path that is visited by very few client IPs 
in the population (this is the reason why it's rare). The client IPs that have 
many interactions with this URI path are anomalous compared to the rest of the 
population that rarely interact with the URI path.

> 

Reviewer: ```suggestion

[discrete]
 [[advanced-jobs]]
 ## Advanced jobs
Advanced jobs give you all the flexibility that's possible in the
create anomaly detection jobs API. At the extreme, you can
switch to directly edit the JSON that will be sent to this endpoint.
All the other types of jobs described in this page _can_ be created
as advanced jobs, but the more specialized wizards make it easier
to create jobs for common situations. You can create an advanced
job if you are familiar with all the functionality that {ml} anomaly
detection provides and want to do something that the more specialized
wizards do not allow you to do.
```

[[ml-anomaly-detection-job-types]]
= {anomaly-detect-cap} job types
++++
<titleabbrev>Job types</titleabbrev>
++++

:keywords: {ml-init}, {stack}, {anomaly-detect}

:description: The description of the different anomaly detection job types. 

{anomaly-jobs-cap} have many possible configuration options which enable you to 
fine-tune the jobs and cover your use case as much as possible. This page 
provides a quick overview of the different types of {anomaly-jobs} and their 
capabilities. The currently available job types are:

* single metric jobs,
* multi-metric jobs,
* population jobs,
* advanced jobs,
* categorization jobs,
* rare jobs.

[discrete]
[[singe-metric-jobs]]
## Single metric jobs

Every {anomaly-job} has at least one detector. A detector defines the type of 

> analysis that occurs (for example, `max`, `average`, or `high`, it's also called 

Author: Here "it's" does not apply to the detector but to the occurring analysis (high, max, etc.). I understand that it might be a source of misunderstanding, so I accept the suggestion. Thanks for spotting the issue.

event is anomalous if the request rate of an IP address is unusually high or low 
compared to the request rate of all IP addresses in the population. The 
population job builds a model of the typical number of requests for the IP 
addresses collectively and compares the behavior of each IP address against that 
collective model to detect outliers. 

[discrete]
[[advanced-jobs]]
## Advanced jobs

Advanced jobs give you all the flexibility that's possible in the 
{ref}/ml-put-job.html[create {anomaly-jobs} API]. At the extreme, you can switch 
to directly edit the JSON that will be sent to this endpoint. All the other 
types of jobs described in this page _can_ be created as advanced jobs, but the 
more specialized wizards make it easier to create jobs for common situations. 
You can create an advanced job if you are familiar with all the functionality 
that {ml} {anomaly-detect} provides and want to do something that the more 
specialized wizards do not allow you to do.

[discrete]
[[categorization-jobs]]
## Categorization jobs

Categorization jobs cluster similar text values together, classify them into 

> categories and detect anomalies within the categories. Categorization works best 

Reviewer 2: Per https://brand.elastic.co/302f66895/p/194a3b-writing-style-guide/t/33fa39 we generally use the serial comma:
```suggestion
categories, and detect anomalies within the categories. Categorization works best 
```

on machine-written text like log messages that typically contains repeated 
strings of text; it does not work well on human-generated text because of its 
high variability.

The model learns the normal volume and pattern of a category over time so the 
job can detect anomalous behavior, such as an unusual number of events in a 
category by using the `count` function or messages that rarely occur by using 
the `rare` function.

[discrete]
[[rare-jobs]]
## Rare jobs

Rare {anomaly-jobs} detect rare occurrences in time series data. Rare jobs use 
the `rare` or `freq_rare` functions and detect such events in populations as 
well. A _rare_ job finds events in simple time series data that occur rarely 
compared to what the model observed over time. A _rare in a population_ job 
finds members of a population that have rare values over time compared to the 
other members of the population. The _frequently rare in a population_ job 

> detects such rare events that frequently occur for a member of a population 

Reviewer 2: I'm a bit confused about what "such" means here. Does it mean "especially rare"? Or is it referring to something from the previous sentence? IMO it should be dropped or clarified:
```suggestion
detects rare events that frequently occur for a member of a population 
```

> compared to other members. As an example, a job models URI paths and client IP 

Reviewer 2: I think this example is related to the last type of rare job, but since three were mentioned in this paragraph, it might be worth saying explicitly. For example:
```suggestion
compared to other members. As an example of this last type of rare job, you can create one that models URI paths and client IP 
```

> interactions and defines a rare URI path that is visited by very few client IPs 

Reviewer 2: Is "defines" the right word here or should it be "detects" or "finds" or "identifies"?
```suggestion
interactions and detects a rare URI path that is visited by very few client IPs 
```
